pinot-core/src/main/java/com/linkedin/pinot/core/periodictask/BasePeriodicTask.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (C) 2014-2018 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.periodictask;
+
+/**
+ * A base class to implement periodic task interface.
+ */
+public abstract class BasePeriodicTask implements PeriodicTask {
+  private final String _taskName;
+  private long _intervalInSeconds;
+  private long _initialDelayInSeconds;
+
+  public BasePeriodicTask(String taskName, long runFrequencyInSeconds, long initialDelayInSeconds) {
+    _taskName = taskName;
+    _intervalInSeconds = runFrequencyInSeconds;
+    _initialDelayInSeconds = initialDelayInSeconds;
+  }
+
+  @Override
+  public void init() {
```

Comment:
Let's remove this empty init() implementation, and always let child classes override this